python/cudf/cudf/core/column_accessor.py

```diff
@@ -0,0 +1,228 @@
+            )
+        return result
+
+    def insert(self, name, value, loc=-1):
+        """
+        Insert value at specified location.
+        """
+        # TODO: we should move all insert logic here
+        if name in self._data:
+            raise ValueError(f"Cannot insert {name}, already exists")
+        if loc == -1 or loc == len(self._data):
+            self._data[name] = value
+        else:
+            name = self._pad_key(name)
+            new_keys = list(self.names)
+            new_values = list(self.columns)
+            new_keys.insert(loc, name)
+            new_values.insert(loc, value)
+            self._data = self._data.__class__(zip(new_keys, new_values),)
+        self._clear_cache()
+
+    def copy(self):
+        return self.__class__(
+            self._data.copy(),
+            multiindex=self.multiindex,
+            level_names=self.level_names,
+        )
+
+    def get_by_label(self, key):
+        if isinstance(key, slice):
+            return self.get_by_label_slice(key)
+        elif isinstance(key, list):
+            return self.__class__(
+                {k: self._data[k] for k in key},
+                multiindex=self.multiindex,
+                level_names=self.level_names,
+            )
+        else:
+            result = self._grouped_data[key]
+            if isinstance(result, cudf.core.column.ColumnBase):
+                return self.__class__({key: result})
+            else:
+                result = _flatten(result)
+                if not isinstance(key, tuple):
+                    key = (key,)
+                return self.__class__(
+                    result,
+                    multiindex=self.nlevels - len(key) > 1,
+                    level_names=self.level_names[len(key) :],
+                )
+
+    def get_by_label_slice(self, key):
+        start = key.start
+        stop = key.stop
+        if start is None:
+            start = self.names[0]
+        if stop is None:
+            stop = self.names[-1]
+        keys = itertools.chain(
+            itertools.takewhile(
+                lambda k: k != stop,
+                itertools.dropwhile(lambda k: k != start, self.keys()),
+            ),
+            (stop,),
+        )
+        return self.__class__(
+            {k: self._data[k] for k in keys},
+            multiindex=self.multiindex,
+            level_names=self.level_names,
+        )
+
+    def get_by_label_with_wildcard(self, key):
+        return self.__class__(
+            {k: self._data[k] for k in self._data if _compare_keys(k, key)},
+            multiindex=self.multiindex,
+            level_names=self.level_names,
+        )
+
+    def get_by_index(self, index):
+        if isinstance(index, slice):
+            start, stop, step = index.indices(len(self._data))
+            keys = itertools.islice(self.keys(), start, stop)
+        elif pd.api.types.is_integer(index):
+            keys = itertools.islice(self.keys(), index, index + 1)
```

Comment:
You're sometimes using `self.keys()` and other times using `self.names` in this function. Should we always use `self.names` to take advantage of the cached property?